src/components/component.ts

```diff
@@ -282,8 +282,6 @@ export class Component {
     }
     this.computeLayout();
     this.render();
-    // flush so that consumers can immediately attach to stuff we create in the DOM
-    RenderController.flush();
```

Comment:
Found a repro:

```Typescript
barplot.renderTo(svg);
console.log(barplot.content().selectAll("rect").size()); // gets 0 instead of the expected value
```